### DISCUSSION TITLE
Saving my ssh passphrase

### DISCUSSION THREAD

**Discussion by tim-kilian** [jesseduffield/lazygit#3062] [General]
I have a passphrase protected ssh key and currently am required to enter the passphrase every time I pull or push to the origin. Can you help me how can I save the passphrase so it wouldn't ask me every time I commit?

**Comment by mark2185** [comment]
Sure! Look into an `SSH agent` program like [keychain](https://wiki.archlinux.org/title/SSH_keys#Keychain).

Once you add your key into the keychain you'll only have to enter the passphrase the first time it is used.

**Reply by tim-kilian** [reply]
Thanks, I think I've got it working with this two stackoverflows:

- https://stackoverflow.com/questions/21965611/how-can-i-run-the-ssh-agent-auto-in-the-zsh-environment
- https://stackoverflow.com/questions/10032461/git-keeps-asking-me-for-my-ssh-key-passphrase

start the agent
`eval $(ssh-agent)`
Add my custom ssh key
`ssh-add ~/.ssh/id_rsa`
Add ssh-agent plugin to my zsh
`plugins=(git ssh-agent)`
Source the config
`source ~/.zshrc`

Works great. Thank you for help, in other editors it is sometimes very painful, when you use a different key than the default one.

**Comment by guttermonk** [comment]
[This thread](https://github.com/jesseduffield/lazygit/discussions/3165) is related and was also very helpful for me.

**Comment by MathiasLui** [comment]
Is there a way to do it without an agent? I don't use one  
It asks me for my SSH key passphrase and I'm just trying to paste it in
